src/ObjectPool/ref/Microsoft.Extensions.ObjectPool.netcoreapp.cs

```diff
@@ -13,6 +13,7 @@ public partial class DefaultObjectPool<T> : Microsoft.Extensions.ObjectPool.Obje
     {
         public DefaultObjectPool(Microsoft.Extensions.ObjectPool.IPooledObjectPolicy<T> policy) { }
         public DefaultObjectPool(Microsoft.Extensions.ObjectPool.IPooledObjectPolicy<T> policy, int maximumRetained) { }
+        [return: System.Diagnostics.CodeAnalysis.NotNullAttribute]
         public override T Get() { throw null; }
         public override void Return(T obj) { }
```

Comment:
Can null be passed to Return? If T is not null everywhere then the notnull generic constraint could be used.

Reply:
`T` already has a `class` constraint. The compiler requires that both `notnull` and `class` to be the first generic constraint so you're unable to apply both simultaneously. That said, there aren't any null checks here. You could return a null value, but that would likely corrupt the pool

Reply:
As I discovered, `T : class` indicates that `T` is non-null. This removes the need to specify a NotNullAttribute, and we get the right behavior